src/Model/ValueObject/String/Behaviour/HasMaxLength.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Model\ValueObject\String\Behaviour;
+
+use CultuurNet\UDB3\Http\ApiProblem\ApiProblem;
+use CultuurNet\UDB3\Http\ApiProblem\SchemaError;
+
+trait HasMaxLength
+{
+    private function hasMaxLength(string $value, int $maxLength, string $path='/'): void
+    {
+        if (mb_strlen($value) > $maxLength) {
+            throw ApiProblem::bodyInvalidData(
```

Comment:
This should not throw an `ApiProblem` because it is not inside the HTTP context, but a `InvalidArgumentException`, also see `guardNotEmpty`.

Reply:
I first did that, but the thing is, if I do it that way I will require a try/catch on every command-handler where this is used for a clean error message.
Now I get all of that for free out of the box. 

Is this not cleaner this way?

Reply:
There is code that catches general or specific exceptions and converts them to an `ApiProblem` to indeed avoid this `try/catch` combination. The code for this can be found inside `ApiProblemFactory::createFromThrowable`. So it's possible to throw an exception here, otherwise all the other guards from the behaviors would have the same issue.

Reply:
I made it throw a custom Exception that I am catching in ApiProblemFactory like discussed